src/Development/Shake/Internal/Rules/File.hs

```diff
@@ -87,16 +94,29 @@ instance BinaryEx [FileQ] where
 instance Hashable FileA where
     hashWithSalt salt (FileA a b c) = hashWithSalt salt a `xor` hashWithSalt salt b `xor` hashWithSalt salt c
 
+instance Hashable FileR where
+    hashWithSalt salt (FileR f _) = hashWithSalt salt f
+
 instance NFData FileA where
     rnf (FileA a b c) = rnf a `seq` rnf b `seq` rnf c
 
+instance NFData FileR where
+    rnf (FileR f _) = rnf f
```

Comment:
NFData should definitely evaluate both pieces - the purpose of NFData is to flush out all exceptions, so everything needs covering.

Reply:
Of course. My bad!